### PR TITLE
Cart Management

### PR DESCRIPTION
**Pull request by dhruv8433**
@pooranjoyb 

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Add product to cart as well as on Supabase [One to Many relationship] #109
Fixes # 109

## Type of change

Please give a X on it which is applicable

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor Code
- [ ] A documentation update
- [ ] Others(mentioned in the issue number)

# How Has This Been Tested?
- when user add any item to cart it also display in `supabase` cart table

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Screenshorts and Vedios:
image

image

# Checklist:
give a X on it which is applicable

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by pooranjoyb — COMMENTED** [review]
Great job, @dhruv8433! 🎉 Thank you for submitting your pull request. Your contribution is valuable and we appreciate your efforts to improve our project.<p>We will promptly review your changes and offer feedback. Keep up the excellent work! Kindly remember to check our [contributing guidelines](https://github.com/pooranjoyb/popShop/blob/dev/CONTRIBUTING.md)</p>

**Comment by dhruv8433** [comment]
@pooranjoyb 

please check this and merge it

**Comment by pooranjoyb** [comment]
Hii @dhruv8433 why aren't you using the `Cart-Slice` we need to store that in store too. Maintain that too.

**Comment by dhruv8433** [comment]
@pooranjoyb 

now you can check it and merge it : 

image

image

**Comment by dhruv8433** [comment]
@pooranjoyb 
please tell me if you want any kind of modifications...

**Comment by pooranjoyb** [comment]
> @pooranjoyb please tell me if you want any kind of modifications...

Can you show me a video demonstration @dhruv8433 

It's not working on my end :/

image

**Comment by dhruv8433** [comment]
Ok i will share you video

**Comment by dhruv8433** [comment]
@pooranjoyb 

please check this and merge it
if you face any issue then inform me

**Comment by pooranjoyb** [comment]
> @pooranjoyb
> 
> please check this and merge it if you face any issue then inform me



> @pooranjoyb
> 
> please check this and merge it if you face any issue then inform me



> @pooranjoyb
> 
> please check this and merge it if you face any issue then inform me

Dude, why `Buy Now` adds to cart. The flow is different. User should go to the `Product-Details` page and `Add to Cart` button should do that logic. Please update it. @dhruv8433

**Comment by dhruv8433** [comment]
@pooranjoyb 
Ok ok 
Got it

**Comment by dhruv8433** [comment]
@pooranjoyb 

now you can check it
image

**Comment by pooranjoyb** [comment]
check build errors by running `npm run build` in you local machine.

**Comment by dhruv8433** [comment]
completed build 
image

**Comment by dhruv8433** [comment]
@pooranjoyb 

now you can check this 
i completed my task as you said

please check this and merge it

**Comment by dhruv8433** [comment]
@pooranjoyb 

deployment and build both are completed 
please merge it

**Comment by dhruv8433** [comment]
@pooranjoyb 

Please when you become free 
Merge it

**Comment by pooranjoyb** [comment]
> @pooranjoyb
> 
> Please when you become free Merge it

Hii @dhruv8433 really sorry, I was occupied few days. Okay I've seen the deployment. Everything is working fine but just one improvement.

- There are static images in the cart. 
- Make sure that, when Product is added to Cart it is should in the Cart page. 
- Use the `slice` or `supabase-table` to render the items added to cart.

That'll do. I'll merge once this part is done. 🚀

**Comment by dhruv8433** [comment]
@pooranjoyb 

i need help when i try to add product for same user it give me `cors error`
can you please tell me how to resolve this

can you allow `cors` from supabase ?

**Comment by pooranjoyb** [comment]
> @pooranjoyb
> 
> i need help when i try to add product for same user it give me `cors error` can you please tell me how to resolve this
> 
> can you allow `cors` from supabase ?

please check in supabase. RLS is already disabled. I guess you already have an access. @dhruv8433

**Comment by dhruv8433** [comment]
@pooranjoyb 

i tried many times but i can't find `allow origin cors` in supabase dashboard
can you please help...

**Comment by dhruv8433** [comment]
@pooranjoyb 

now everything working as expected
please check this and merge it

image

output:

image

**Comment by dhruv8433** [comment]
@pooranjoyb 

https://github.com/pooranjoyb/popShop/assets/114583978/7f2a720e-871e-4a1e-9739-5166d3805293

**Comment by dhruv8433** [comment]
@pooranjoyb 

now you can merge it
please check everything working as expected

**Comment by pooranjoyb** [comment]
Awesome work @dhruv8433 🚀 
What was the issue with `cors` tho ?

File: src/utils/cors.ts
Comment (pooranjoyb) [inline src/utils/cors.ts:1]: Did you use this file somewhere? @dhruv8433

**Review by pooranjoyb — APPROVED** [review]

**Comment by dhruv8433** [comment]
@pooranjoyb 

no i just redisable RLS and try than it worked

**Comment by dhruv8433** [comment]
please merge it

**Comment by pooranjoyb** [comment]
> please merge it

Dude it ain't working, just checked the deployment. :/

[screen-recorder-sun-jun-09-2024-18-59-44.webm](https://github.com/pooranjoyb/popShop/assets/90945182/0b526c5b-09f0-4717-8988-97c019719bda)

**Comment by pooranjoyb** [comment]
Please lmk if I'm missing something, do create a new record if the user doesnot exist in the table. @dhruv8433

**Review by pooranjoyb — APPROVED** [review]

**Comment by dhruv8433** [comment]
@pooranjoyb 

just wait i provide you a video

**Comment by pooranjoyb** [comment]
A hell loota work, init @dhruv8433 ? 😆 
But yeah at the end its working. So, well done! 💯

**Comment by dhruv8433** [comment]
Thank you @pooranjoyb 

idk why video not downloaded bu thanks again

**Comment by pooranjoyb** [comment]
> Thank you @pooranjoyb
> 
> idk why video not downloaded bu thanks again

But yeah there is still one issue. I guess that's probably easy to work around. 
- The functioning is working properly in a single browser but not in other browsers as you are fetching data from redux store.
- while rendering the cart items fetch it from db and show. 
- Although the admin has two items in cart its not showing up in carts page.

@dhruv8433 

you can work on this in a separate issue. 👍🏼

**Comment by dhruv8433** [comment]
@pooranjoyb 
Ok i will create separate issue for thay

**Pull request merged**